pythonforandroid/graph.py

```diff
@@ -7,49 +7,79 @@
 from pythonforandroid.util import BuildInterruptingException
 
 
-class RecipeOrder(dict):
+def fix_deplist(deps):
+    """ Turn a dependency list into lowercase, and make sure all entries
+        that are just a string become a tuple of strings
+    """
+    deps = [
+        ((dep.lower(),) if not isinstance(dep, (list, tuple))
+         else tuple([dep_entry.lower() for dep_entry in dep]))
+        for dep in deps
+    ]
+    return deps
```

Comment:
Minor: these isolated utils functions are easy to unit test. Even if they may look trivial, I think next time they deserve some tests, what do you think? Please skip it for now, but I'll slowly try to increase the coverage of p4a and may introduce some coverage check at some point :smiling_imp:

Reply:
Added a unit test! :+1: